sql/core/src/main/java/org/apache/spark/sql/execution/datasources/orc/OrcColumnarBatchReader.java

```diff
@@ -186,7 +188,12 @@ public void initBatch(
         int colId = requestedDataColIds[i];
         // Initialize the missing columns once.
         if (colId == -1) {
-          OnHeapColumnVector missingCol = new OnHeapColumnVector(capacity, dt);
+          final WritableColumnVector missingCol;
```

Comment:
Is it possible to use `ConstantColumnVector` for the `missingCol`?

Reply:
It seems simpler to use `ConstantColumnVector` here. I've updated the PR